python/pyspark/sql/datasource.py

```diff
@@ -357,6 +363,30 @@ def abort(self, messages: List["WriterCommitMessage"]) -> None:
         ...
 
 
+class DataStreamOffset:
+    def __init__(self, value: Any) -> None:
+        self.value = value
+
+    def set_value(self, json_str: str) -> None:
+        self.value = self.from_json(json_str)
+
+class DataStreamReader(ABC):
+    def latest_offset(self) -> DataStreamOffset:
```

Comment:
By design, all Python API follows `camelCase` naming rule. If we could make it one word like `latestoffset`, that's an option too (to make it more Pythonic)

Reply:
A rather minor but pertinent point

I see

from pyspark.sql.session import SparkSession

or equally

from pyspark.sql import SparkSession

Although they are both valid, in practice, many developers prefer the second option (from pyspark.sql import SparkSession) as it is concise and aligns with the common convention of grouping related functionality under a common namespace or module. It also reflects the structure of the pyspark.sql module, where SparkSession is a key component.